src/dscanner/analysis/cyclomatic_complexity.d

```diff
@@ -0,0 +1,260 @@
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+module dscanner.analysis.cyclomatic_complexity;
+
+import dparse.ast;
+import dparse.lexer;
+import dsymbol.scope_ : Scope;
+import dscanner.analysis.base;
+import dscanner.analysis.helpers;
+
+import std.format;
+
+/// Implements a basic cyclomatic complexity algorithm using the AST.
+///
+/// Issues a warning on functions whenever the cyclomatic complexity of them
+/// passed over a configurable threshold.
+///
+/// The complexity score starts at 1 and is increased each time on
+/// - `if`
+/// - switch `case`
+/// - any loop
+/// - `&&`
+/// - `||`
+/// - `?:` (ternary operator)
+/// - `throw`
+/// - `catch`
+/// - `return`
```

Comment:
Giving `return` a negative cost means that pyramid-of-doom style has a better score than early-return style :) But I guess that would be offset by some maximum-nesting check or something.